Hub-Server/src/main/java/edu/yu/cs/hub/Hub.java

```diff
@@ -93,4 +106,10 @@ private void postToGallery(URL url) {
                 .bodyToMono(String.class).block();
     }
     
+    //Logic for setting leader election. Currently, just selects the first-added gallery (id of 1). Can be modified in the future.
+    private void setLeader(GalleryInfo gi) {
```

Comment:
I must have misunderstood your approach. I thought you meant the first gallery to register itself

What if the gallery with id of 1 is not up?

Reply:
You're totally right. We were thinking about that, but held off from implementing because it works with the notion of "pinging" the current leader server which you mentioned we shouldn't be doing yet. 

This is what the method really should look like:
        if (galleryInfoRepo.getLeader() == null || <method to ping the current leader server and see if it is up>) {
            galleryInfoRepo.makeLeader(gi)
        }

The first part of the OR statement is just a bit more dyanmic way to check if a leader was never set (which 99% of the time should default to ID of 1). 
The second part is where we would ping the current leader. I imagine, this would look like making a new class and endpoint that would just return a 200, or something like that. If the call returns OK, we know the leadedr is up and keep it. If not, we make a new leader (and swtich the flag off for the old leader).

This check would now happen both when a gallery is first posted, and when the galleries register with the hub on startup in the PUT method.

We could do this know, but it seems like that notion of pinging the leader server is what you're introducing in the next phase, where we can do it the "right" way. 

Can we wait until after the merge to address this or should we do something now?

Reply:
Lets just address my second question for now - the stress on server 1. I'd like to find a better strategy that will mix things up between servers. 

Pinging the leader will be for the next stage.

Reply:
We reorganized and made a serverstartup and utility class within the hub-server side of the project. on serverstartup and also whenever a gallery is created, the hub randomly picks a leader and sends that information to all of the servers.